src/main/java/fr/lesprojetscagnottes/core/slack/controller/SlackTeamController.java

```diff
@@ -69,22 +70,67 @@
+            LOGGER.error("Impossible to get Slack team by ID : principal has not enough privileges");
+            throw new ForbiddenException();
+        }
+
+        // Transform and return organization
+        return SlackTeamModel.fromEntity(entity);
+    }
+
+
+    @Operation(summary = "Update a Slack team", description = "Update a Slack team", tags = {"Team:Slack"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Return the content", content = @io.swagger.v3.oas.annotations.media.Content(schema = @Schema(implementation = ContentModel.class))),
+            @ApiResponse(responseCode = "400", description = "Some references are missing", content = @io.swagger.v3.oas.annotations.media.Content(schema = @Schema())),
+            @ApiResponse(responseCode = "403", description = "Some references doesn't exist", content = @Content(schema = @Schema())),
+            @ApiResponse(responseCode = "404", description = "Principal has not enough privileges", content = @Content(schema = @Schema()))
+    })
+    @PreAuthorize("hasRole('USER')")
+    @RequestMapping(value = "/team/slack", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public SlackTeamModel update(Principal principal, @RequestBody SlackTeamModel slackTeamModel) {
+
+        // Verify that ID is correct
+        // Fails if any of references are null
+        if (slackTeamModel == null || StringUtils.isEmpty(slackTeamModel.getTeamId()) || StringUtils.isEmpty(slackTeamModel.getPublicationChannelId())) {
+            if (slackTeamModel != null) {
+                log.error("Impossible to update project {} : some references are missing", slackTeamModel.getId());
+            } else {
+                log.error("Impossible to update a null project");
+            }
+            throw new BadRequestException();
+        }
+        // Verify that entity exists
+        SlackTeamEntity entity = slackTeamRepository.findByTeamId(slackTeamModel.getTeamId());
+        if (entity == null) {
             LOGGER.error("Impossible to get Slack team by ID : Slack team not found");
             throw new NotFoundException();
+        } else {
+            entity.setPublicationChannelId(slackTeamModel.getPublicationChannelId());
+            entity.setTeamName(slackTeamModel.getTeamName());
+            SlackTeamModel.fromEntity(slackTeamRepository.save(entity));
         }
 
         // If user is not admin => organization where principal is member
         // Else => all organizations
         Long userLoggedInId = userService.get(principal).getId();
-        if(!userService.isMemberOfOrganization(userLoggedInId, entity.getOrganization().getId()) && userService.isNotAdmin(userLoggedInId)) {
+        if (!userService.isMemberOfOrganization(userLoggedInId, entity.getOrganization().getId()) && userService.isNotAdmin(userLoggedInId)) {
```

Comment:
Permission is incorrect. To modify a SlackTeam, the user must be a manager of the organization.

You should use `userService.isManagerOfOrganization(...)` method

Reply:
considered